building-an-opengraph-image-generation-api-with-cloudinary-netlify-functions-and-react/README.md

```diff
@@ -0,0 +1,39 @@
+## About this course
+
+**📝Important Note:** In order to get the OpenGraph Image working in Netlify you need [to change the Build image](https://docs.netlify.com/configure-builds/get-started/#build-image-selection) from &ldquo;Ubuntu Xenial 16.04&rdquo;, which is the default, to &ldquo;Ubuntu Trusty 14.04&rdquo;. Since this collection was published something has changed with Netlify's Build Machines and the original code no longer works on Netlify's Machines, it does, however still work on the Legacy Machines
```

Comment:
This is strange. I wonder what changed 🤔  Both my production instance of this code and the one I deployed for the course are on xenial.

Do you happen to remember what error you ran into?

Reply:
I had done this collection a while ago and it had just worked, but when I refreshed my image.js when I was going through the collection again, and my site rebuilt I ran into this error when I ran the `gen-opengraph-image` function (I was able to retrieve it from a previous Netlify build):

``` json
{
  "errorType": "Error",
  "errorMessage": "ENOENT: no such file or directory, open './image.js'",
  "trace": [
    "Error: ENOENT: no such file or directory, open './image.js'",
    "    at Object.openSync (fs.js:458:3)",
    "    at Object.readFileSync (fs.js:360:35)",
    "    at Object.&lt;anonymous&gt; (/var/task/src/gen-opengraph-image.js:3:19)",
    "    at Module._compile (internal/modules/cjs/loader.js:1138:30)",
    "    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1158:10)",
    "    at Module.load (internal/modules/cjs/loader.js:986:32)",
    "    at Function.Module._load (internal/modules/cjs/loader.js:879:14)",
    "    at Module.require (internal/modules/cjs/loader.js:1026:19)",
    "    at require (internal/modules/cjs/helpers.js:72:18)",
    "    at Object.&lt;anonymous&gt; (/var/task/gen-opengraph-image.js:1:18)"
  ]
}
```

Reply:
It seemed to be something related to this call: https://github.com/ChristopherBiscardi/egghead-opengraph-images/blob/7f0bf4affd4da0ac57d3e83988e5aa911c9ba82a/functions/gen-opengraph-image/gen-opengraph-image.js#L3

But the build completed successfully and Rollup reported during the build process that the image.js file was created. I tried to troubleshoot for a good bit, but then tried switching the build image and that worked so 🤷🏼‍♂️

Reply:
ah yes, netlify changed something in their handing of execution so that __dirname and such don't resolve as they used to, as described here: https://twitter.com/danestves/status/1287522369590919168